File: 509-fibonacci-number/509-fibonacci-number.java
class Solution {
    int cache[] = new int[100];
    public int fib(int n) {
       
        if (n <= 1) {
            return n;
        }
        if (cache[n-1] == 0) {
            cache[n-1] = fib(n-1);
        }
        if (cache[n-2] == 0) {
            cache[n-2] = fib(n-2);
        }
        return cache[n-1]+cache[n-2];
    }
}

File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n == 0)
            return 0;
        if(n <= 2)
            return 1;
        int a = 1;
        int b = 1;
        n-=2;
        while(n)
        {
            b = b + a;
            a = b - a;
            n--;
        }
        return b;
    }
};